11/umbraco-cms/extending/packages/good-practice-and-defaults.md

```diff
@@ -0,0 +1,207 @@
+---
+description: Information on good practices and common defaults for Umbraco package development.
+---
+
+# Good practice and defaults
+
+This document provides guides and notes on package development. It includes good practice guidelines that will help you maintain and support your package through multiple releases and versions of Umbraco. These good practices are not prescriptive, but offer a guide as to what often works well, and not-so-well, when developing packages for Umbraco.
+
+## App Plugins
+
+To extend the Umbraco backoffice, a package can provide files such as a `package.manifest` and AngularJS views/controllers that should be stored within the `App_Plugins` folder. It's recommended to put all files in a subfolder with a unique name, preferably using the package name, like `App_Plugins\MyPackage`.
+
+Files in the `App_Plugins` folder will be publicly available on the website even though they are not in the root folder. You should not store sensitive information in the `App_Plugins` folder.
+
+Files in the `App_Plugins` folder should be considered immutable. This means that they are not something a user of your package is expected to change on their site.
+
+The default delivery method for files to the `App_Plugins` folder is via a `.targets` file within a package. This means when a website is built, the files in this folder are copied over from the NuGet cache. When this happens, any changes a user might have made to these files will be lost. Equally, if the user performs a `dotnet clean` on a solution, all files in the `App_Plugins` folder will be deleted.
+
+If you have files that you expect users of your package to alter you should not place them in the `App_Plugins` folder.
+
+{% hint style="info" %}
+The `App_Plugins` folder does not exist on a fresh installation of Umbraco. You need to create it manually before you save your license file to this folder.
+{% endhint %}
```

Comment:
Looks like the hint was incorrectly copied from the license section, so we can either remove it:
```suggestion
If you have files that you expect users of your package to alter you should not place them in the `App_Plugins` folder.
```

And optionally markup this last sentence as hint:
```suggestion
{% hint style="info" %}
If you have files that you expect users of your package to alter you should not place them in the `App_Plugins` folder.
{% endhint %}
```